### PR TITLE
Implement ButtonsPlacement property to have possibility to set NUD on the left side

### PR DESCRIPTION
**Pull request by QuantumDeveloper**

### CODE REVIEW

File: MahApps.Metro/Controls/NumericUpDown.cs [hunk 1]
```diff
@@ -70,7 +70,13 @@ private static void IsReadOnlyPropertyChangedCallback(DependencyObject dependenc
             typeof(NumericUpDown),
             new FrameworkPropertyMetadata(default(double?), FrameworkPropertyMetadataOptions.BindsTwoWayByDefault, OnValueChanged, CoerceValue));
 
-        public static readonly DependencyProperty MinimumProperty = DependencyProperty.Register(
+      public static readonly DependencyProperty ButtonsPlacementProperty = DependencyProperty.Register(
+           "ButtonsPlacement",
+           typeof(Dock),
```
Comment (punker76) [inline MahApps.Metro/Controls/NumericUpDown.cs]: Can you create an enumeration for this, instead using `Dock`?

something like

```csharp
enum ButtonAlignment
{
    Left,
    Right
}
```

i think all other values make no sense

File: MahApps.Metro/Controls/NumericUpDown.cs [hunk 2]
```diff
@@ -70,7 +70,13 @@ private static void IsReadOnlyPropertyChangedCallback(DependencyObject dependenc
...
+           typeof(NumericUpDown),
+           new FrameworkPropertyMetadata(Dock.Right, FrameworkPropertyMetadataOptions.BindsTwoWayByDefault));
+
+      public static readonly DependencyProperty MinimumProperty = DependencyProperty.Register(
```
Comment (punker76) [inline MahApps.Metro/Controls/NumericUpDown.cs]: can you follow MahApps' coding styleguide?

**Comment by punker76** [comment]
@QuantumDeveloper two notes...
1) why changing the buttons from "+ -" to "- +"
2) the ClearTextButton is still right...
image

**Comment by QuantumDeveloper** [comment]
1. I made mirroring of the buttons to make it symmetric. If you want, I can change order.
2. Need to deal with it.

**Comment by thoemmi** [comment]
Wouldn't be `ButtonsPlacement` a better name than `ButtonsPlacementVariant`? /cc @punker76

**Comment by punker76** [comment]
@thoemmi what about `Location` for the enumeration and `ButtonLocation` for the dp? (about how xceed does it)

**Comment by thoemmi** [comment]
Maybe `ButtonsAlignment` is more consistent with WPF's naming. And `...Buttons` should be plural, because it refers to all buttons of a TextBox.

**Comment by punker76** [comment]
@thoemmi yeah, i forgot the `s` and i am agreed with `ButtonsAlignment`... for both? enum and dp?

**Comment by thoemmi** [comment]
Yes, I'd call both the enum and the dp `ButtonsAlignment`.

**Comment by punker76** [comment]
@thoemmi done 66e26652c249717ede763ae5cc842be3d78579f7

**Comment by thoemmi** [comment]
:+1:

**Pull request merged**